### PR TITLE
BE-006-1 (Nikolay Chornyi) Add entity, DTO

### PR DESCRIPTION
**Pull request by mblack666**
Add entities, DTOs & update table discounts

### CODE REVIEW

File: src/main/java/com/exadel/discount/entity/Vendor.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import java.util.Set;
+import java.util.UUID;
+
+@Data
+@NoArgsConstructor
+@EqualsAndHashCode(exclude = {"vendorLocations"})
+@ToString(exclude = {"vendorLocations"})
```
Comment (el-dub) [inline src/main/java/com/exadel/discount/entity/Vendor.java]: Add discounts

File: src/main/java/com/exadel/discount/entity/Discount.java [hunk 2]
```diff
@@ -15,32 +14,53 @@
...
 
-@Entity
-@Table(name = "discounts")
 @Data
 @NoArgsConstructor
-@AllArgsConstructor
+@Entity
+@Table(name = "discounts")
 @EqualsAndHashCode(exclude = {"tags"})
 @ToString(exclude = {"tags"})
```
Comment (el-dub) [inline src/main/java/com/exadel/discount/entity/Discount.java]: Add vendorLocations

File: src/main/java/com/exadel/discount/entity/Vendor.java (hunk 1 above)
Comment (mblack666) [reply src/main/java/com/exadel/discount/entity/Vendor.java]: +

File: src/main/java/com/exadel/discount/entity/Discount.java (hunk 2 above)
Comment (mblack666) [reply src/main/java/com/exadel/discount/entity/Discount.java]: +

**Review by el-dub — DISMISSED** [review]

File: src/main/resources/db/changelog/v.0.3/db.changelog-add-vendor-to-discount.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+databaseChangeLog:
+  - changeSet:
+      id: 37-add-vendor-to-dixount
```
Comment (Hanna-Kalashnyk) [inline src/main/resources/db/changelog/v.0.3/db.changelog-add-vendor-to-discount.yaml]: is here no typo in word  "dixount" ?

**Review by Hanna-Kalashnyk — DISMISSED** [review]

File: src/main/resources/db/changelog/v.0.3/db.changelog-add-vendor-to-discount.yaml (hunk 3 above)
Comment (mblack666) [reply src/main/resources/db/changelog/v.0.3/db.changelog-add-vendor-to-discount.yaml]: +

File: src/main/java/com/exadel/discount/dto/vendor/CreateVendorDTO.java [hunk 4]
```diff
@@ -1,21 +1,16 @@
...
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
 
 @Data
-@Builder
-@AllArgsConstructor
-@NoArgsConstructor
 public class CreateVendorDTO {
     @NotBlank
+    @Size(max = 50)
```
Comment (el-dub) [inline src/main/java/com/exadel/discount/dto/vendor/CreateVendorDTO.java]: Add messages

Comment (mblack666) [reply src/main/java/com/exadel/discount/dto/vendor/CreateVendorDTO.java]: +

**Review by el-dub — DISMISSED** [review]

File: src/main/java/com/exadel/discount/controller/VendorController.java [hunk 5]
```diff
@@ -19,72 +18,12 @@
 @RestController
 @RequestMapping("/vendors")
 public class VendorController {
```
Comment (mikevoloshyn) [inline src/main/java/com/exadel/discount/controller/VendorController.java]: empty controller

File: src/main/java/com/exadel/discount/dto/CategoryDTO.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import javax.validation.constraints.Null;
+import java.util.UUID;
+
+@Data
+public class CategoryDTO {
+    @Null(groups = Create.class, message = "Сategory ID should be blank")
+    private UUID id;
+    @NotBlank(groups = Create.class, message = "Category name should be not blank")
+    private String name;
+
```
Comment (mikevoloshyn) [inline src/main/java/com/exadel/discount/dto/CategoryDTO.java]: empty line

File: src/main/java/com/exadel/discount/dto/location/CreateLocationDTO.java [hunk 7]
```diff
@@ -1,22 +1,17 @@
...
-    @NotBlank
-    private String country;
-    @NotBlank
-    private String city;
-    @NotBlank
-    private String address;
+    //TODO
+    /*private CountryDTO country;
+    private CityDTO city;*/
     private String contact;
```
Comment (mikevoloshyn) [inline src/main/java/com/exadel/discount/dto/location/CreateLocationDTO.java:10]: please remove todo lines

**Review by mikevoloshyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/exadel/discount/dto/location/CreateLocationDTO.java (hunk 7 above)
Comment (mblack666) [reply src/main/java/com/exadel/discount/dto/location/CreateLocationDTO.java:10]: +

File: src/main/java/com/exadel/discount/dto/CategoryDTO.java (hunk 6 above)
Comment (mblack666) [reply src/main/java/com/exadel/discount/dto/CategoryDTO.java]: +

**Review by el-dub — APPROVED** [review]

**Review by mikevoloshyn — APPROVED** [review]

**Review by mikevoloshyn — APPROVED** [review]

**Review by rudik-byte — APPROVED** [review]

**Review by SerhiiPakulia — APPROVED** [review]

**Pull request merged**